lib/ansible/modules/packaging/language/yarn.py

```diff
@@ -0,0 +1,316 @@
+
+    def install(self):
+        # Yarn has 2 separate install commands: one for specific packages and oen that installs
+        # all packages from package.json.
+        if self.name or self.name_version:
+            return self._exec(['add'])
+        return self._exec(['install'])
+
+    def update(self):
+        return self._exec(['upgrade'])
+
+    def uninstall(self):
+        return self._exec(['remove'])
+
+    def list_outdated(self):
+        outdated = list()
+        data = self._exec(['outdated'], True, False)
+        for dep in data.splitlines():
+            if dep:
+                # node.js v0.10.22 changed the `npm outdated` module separator
+                # from "@" to " ". Split on both for backwards compatibility.
+                pkg, other = re.split('\s|@', dep, 1)
+                outdated.append(pkg)
+
+        return outdated
+
+
+def main():
+    arg_spec = dict(
+        name=dict(default=None),
+        path=dict(default=None, type='path'),
+        version=dict(default=None),
+        production=dict(default='no', type='bool'),
+        executable=dict(default=None, type='path'),
+        registry=dict(default=None),
+        state=dict(default='present', choices=['present', 'absent', 'latest']),
+        ignore_scripts=dict(default=False, type='bool'),
+    )
+    arg_spec['global'] = dict(default='no', type='bool')
+    module = AnsibleModule(
+        argument_spec=arg_spec,
+        supports_check_mode=True
+    )
+
+    name = module.params['name']
+    path = module.params['path']
+    version = module.params['version']
+    glbl = module.params['global']
+    production = module.params['production']
+    executable = module.params['executable']
+    registry = module.params['registry']
+    state = module.params['state']
+    ignore_scripts = module.params['ignore_scripts']
+
+    if not path and not glbl:
+        module.fail_json(msg='Path must be specified when not using global arg')
+    if state == 'absent' and not name:
+        module.fail_json(msg='Uninstalling a package is only available for named packages')
+
+    yarn = Yarn(module, name=name, path=path, version=version, glbl=glbl, production=production, \
+              executable=executable, registry=registry, ignore_scripts=ignore_scripts)
+
+    changed = False
+    if state == 'present':
+        installed, missing = yarn.list()
+        if len(missing):
+            changed = True
+            out = yarn.install()
+    elif state == 'latest':
+        installed, missing = yarn.list()
+        outdated = yarn.list_outdated()
+        if len(missing):
+            changed = True
+            out = yarn.install()
+        if len(outdated):
+            changed = True
+            out = yarn.update()
+    else: #absent
+        installed, missing = yarn.list()
+        if name in installed:
+            changed = True
+            out = yarn.uninstall()
+
+    module.exit_json(changed=changed, out=out)
+
+# import module snippets
```

Comment:
We recommend not `import *` and stylistically prefer `from ansible.module_utils.basic import AnsibleModule` to come right after the global docstrings.